addons/vehicles/XEH_preInit.sqf

```diff
@@ -6,4 +6,37 @@ PREP_RECOMPILE_START;
 #include "XEH_PREP.hpp"
 PREP_RECOMPILE_END;
 
+#include "initSettings.sqf"
+
+[
+    "AllVehicles",
+    "initPost",
+    {
+        params ["_vehicle"];
+        _vehicle setVariable [QGVAR(engineState), isEngineOn _vehicle];
+    }
+] call CBA_fnc_addClassEventHandler;
+[
+    "AllVehicles",
+    "engine",
+    {
+        params ["_vehicle", "_state"];
+        _vehicle setVariable [QGVAR(engineState), _state];
+    }
+] call CBA_fnc_addClassEventHandler;
+[
+    "AllVehicles",
+    "getout",
+    {
+        params ["_vehicle", "_position", "_unit"];
+        if (!GVAR(keepEngineRunning)) exitWith {};
+        if (_unit == ACE_player && {_position == "driver"} && {_vehicle getVariable [QGVAR(engineState), false]}) then {
```

Comment:
```suggestion
        if (_unit == ACE_player && {_position == "driver" && {_vehicle getVariable [QGVAR(engineState), false]}}) then {
```
optimize

Reply:
isEqualTo on the string? Not sure where the position string comes from and if casing is guaranteed.

Reply:
I don't trust BI not the change the capitalization of this.